[90X] Replace TMatrix by Eigen matrices and various fixes in MillePede workflow

ghellwig opened the pull request:
backport of #18177

This PR introduces the Eigen-based version of GBL and removes further usage of TMatrix from th MillePede workflow.
The runtime of the Mille step is reduced by  approx. 30% without loss of generality and without changing the functionality.
Further changes only improve user-friendliness by adding better examples, protections and examples.

The changes of this PR are based on #18139.

gregor-mittag committed:
Move to Eigen-version of GBL developed by Claus Kleinwort.

gregor-mittag committed:
Minimal adaption of CMSSW code to deal with new GBL Eigen interfaces.

gregor-mittag committed:
Enable ROOT support in GBL-eigen.

gregor-mittag committed:
Removed TMatrix from MillePedeAlignmentAlgorithm.

 - compiles and is technically working
 - need to check the correctness of the output
 - need to optimize usage regarding fixed sizes

gregor-mittag committed:
Optimize usage of Eigen matrices in 'MillePedeAlignmentAlgorithm'.

gregor-mittag committed:
Removed TMatrix from ReferenceTrajectories.

 - compiles and is technically working
 - need to check the correctness of the output

gregor-mittag committed:
Create common base class for alignment producers.

gregor-mittag committed:
Derive ESProducerLooper-version of alignment producer from common base class.

gregor-mittag committed:
Removed `EventSetup` from `endOfLoop` methods in alignment monitors.

 - needed for unification of alignment producers
 - in addition:
   - `EventSetup` was nowhere used
   - in case of zero processed events it would be undefined

gregor-mittag committed:
Derive EDAnalyzer-version of alignment producer from common base class.

gregor-mittag committed:
Added documentation to alignment producer base class.

gregor-mittag committed:
Make raw pointer to `AlignmentAlgorithmBase` `std::unique_ptr`.

gregor-mittag committed:
Make raw pointers to `AlignmentMonitorBase` `std::unique_ptr`.

gregor-mittag committed:
Make `globalPositions_` `std::unique_ptr`.

gregor-mittag committed:
Renamed `PCLTrackerAlProducer` to more generic `AlignmentProducerAsAnalyzer`.

gregor-mittag committed:
Make `AlignmentProducerAsAnalyzer` an `edm::one` module.

gregor-mittag committed:
Renamed python config from `TrackerAlignmentProducerForPCL_cff` to generic `AlignmentProducerAsAnalyzer_cff`.

gregor-mittag committed:
Make test config of PCL workflow geometry-independent.

gregor-mittag committed:
Migrate offline MillePede workflow to analyzer-version of the alignment producer.

gregor-mittag committed:
Clean up of pede shell script + usage of parallel binary copy.

gregor-mittag committed:
Optimize 'GblData'.

gregor-mittag committed:
Optimize 'GblPoint'.

gregor-mittag committed:
Optimize 'GblTrajectory'.

gregor-mittag committed:
Replace fixed-length std::vector with std::array.

gregor-mittag committed:
Replace copy into container by move into container.

gregor-mittag committed:
Optimize usage of 'proDer' matrices in 'GblTrajectory'.

gregor-mittag committed:
Remove tabs and trailing whitespace. Purely cosmetic and no functional changes.

gregor-mittag committed:
Avoid redundant step in 'GblPoint.h'.

gregor-mittag committed:
Move typedefs in gbl namespace.

gregor-mittag committed:
Replace call-by-value with call-by-const-reference.

gregor-mittag committed:
Code clean-up.

gregor-mittag committed:
Turned off GBL_EIGEN_SUPPORT_ROOT.

gregor-mittag committed:
Fix usage of payloads from 'pro' database.

gregor-mittag committed:
Fix missing replacement of 'TTRHBuilder'.

gregor-mittag committed:
Automatically turn off pixel quality flag if templates are not used.

gregor-mittag committed:
Cosmetic changes to the unified sequence.

gregor-mittag committed:
Added 'TTRHBuilder' option to 'MilleSetup'.

gregor-mittag committed:
Include required offline beam spot module in unified refitting sequence.

gregor-mittag committed:
Use appropriate default value corresponding to phase-1 GT.

gregor-mittag committed:
Support auto GT in db-query tools.

gregor-mittag committed:
Added unit test for MPS workflow.

gregor-mittag committed:
Add option to redirect mille binaries to another mass storage directory.

 - add test mode to redirect it to a dummy path
   -> avoids eos command which depends on network connection

gregor-mittag committed:
Allow 'cmsexpress' as a mille queue.

gregor-mittag committed:
Allow more flexibility for single IOV output.

gregor-mittag committed:
Better handling of subprocess and system calls in `mps_alisetup.py`.

 - exit with meaningful error message in case of failure
 - previously some failures where silently ignored leading to
   confusing failures later

gregor-mittag committed:
Add utility to easily add EDFilter to mille step.

gregor-mittag committed:
Add APV mode filter to ensure consistency of cosmics configuration.

gregor-mittag committed:
Adjust mps template examples to Phase-1.

Added also a full module-level alignment example.

gregor-mittag committed:
Remove debug remnant.

gregor-mittag committed:
Fix typo in file list creator.

ghellwig commented:
please test

cmsbuild commented:
A new Pull Request was created by  @ghellwig (Gregor Mittag) for CMSSW_9_0_X.

It involves the following packages:

Alignment/CommonAlignment
Alignment/CommonAlignmentMonitor
Alignment/CommonAlignmentProducer
Alignment/MillePedeAlignmentAlgorithm
Alignment/OfflineValidation
Alignment/ReferenceTrajectories
Alignment/TrackerAlignment


@ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @mmusich, @davidlange6 can you please review it and eventually sign? Thanks.
@ghellwig, @pakhotin, @tocheng, @tlampen, @mschrode, @mmusich this is something you requested to watch as well.
@davidlange6, @smuzaffar you are the release manager for this.

cms-bot commands are listed here cms-sw/cmssw#13028

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: de2a302a9a5c645f64f0fa0e44306a6f12bd4f52
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/18871/summary.html

cmsbuild commented:
Comparison job queued.

gregor-mittag committed:
Fix clang warning in 'GblTrajectory.cc'.

$CMSSW_BASE/src/Alignment/ReferenceTrajectories/src/GblTrajectory.cc:1045:36: warning: moving a temporary object prevents copy elision [-Wpessimizing-move]
        innerTransDer.emplace_back(std::move(
                                   ^
$CMSSW_BASE/src/Alignment/ReferenceTrajectories/src/GblTrajectory.cc:1045:36: note: remove std::move call here
        innerTransDer.emplace_back(std::move(
                                   ^~~~~~~~~~

gregor-mittag committed:
Comply to CMSSW coding rules.

The try-catch clause is redundant here because the only exception
which could be thrown is if the memory allocation for the FileBlob
content does not succeed, i.e. we are out of RAM.
If this happens, the job should quit anyways, I think.

Every other failure is covered by the check for the FileBlob size.

ghellwig commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #18178 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @mmusich, @davidlange6 can you please check and sign again.

cmsbuild commented:
+1
Tested at: be4980491073382716fa16919d67f97ce5a04f33
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/18888/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/18888/summary.html

gregor-mittag committed:
Fix bug in covariance diagonalisation.

cmsbuild commented:
Pull request #18178 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @mmusich, @davidlange6 can you please check and sign again.

ghellwig commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 2f77dbdd0c1320d0ec957a42e930e1347a134a21
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/18946/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/18946/summary.html

gregor-mittag committed:
Fix 'create_single_iov_db' method for local sqlite files.

ghellwig commented:
please test

cmsbuild commented:
Pull request #18178 was updated. @ghellwig, @arunhep, @cerminar, @cmsbuild, @franzoni, @mmusich, @davidlange6 can you please check and sign again.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f1440f81633fa24e88b404e1cd8e2af62244f990
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19064/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19064/summary.html

Comparison Summary:
 - You potentially added 3 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1915960
 - DQMHistoTests: Total failures: 14967
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1900820
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files

arunhep commented:
please test
comparisons are no longer available

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: f1440f81633fa24e88b404e1cd8e2af62244f990

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19358/summary.html

I found follow errors while testing this PR

**Failed tests:  AddOn**

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PRef --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PRef_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 00:24:15 2017-date Tue Apr 25 00:18:18 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 00:24:15 2017-date Tue Apr 25 00:18:18 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 00:24:15 2017-date Tue Apr 25 00:18:18 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PIon --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PIon_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 00:26:51 2017-date Tue Apr 25 00:18:47 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 00:26:51 2017-date Tue Apr 25 00:18:47 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 00:26:51 2017-date Tue Apr 25 00:18:47 2017 s - exit: 21504

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19358/summary.html

There are some workflows for which there are errors in the baseline:
 1306.0 step 1
 The results for the comparisons for these workflows could be incomplete 
This means most likely that the IB is having errors in the relvals.The error does NOT come from this pull request

Comparison Summary:
 - You potentially added 58 lines to the logs
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 22
 - DQMHistoTests: Total histograms compared: 1735705
 - DQMHistoTests: Total failures: 29896
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1705646
 - DQMHistoTests: Total skipped: 163
 - DQMHistoTests: Total Missing objects: 0
 - Checked 90 log files, 14 edm output root files, 22 DQM output files

ghellwig commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
-1

Tested at: f1440f81633fa24e88b404e1cd8e2af62244f990

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19362/summary.html

I found follow errors while testing this PR

**Failed tests:  RelVals  AddOn**

* **RelVals**:

When I ran the RelVals I found an error in the following worklfows:
4.53 step2
<pre>runTheMatrix-results/4.53_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT/step2_RunPhoton2012B+RunPhoton2012B+HLTD+RECODR1reHLT+HARVESTDR1reHLT.log

</pre>

* **AddOn**:

I found errors in the following addon tests:

cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PRef --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PRef_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 08:50:40 2017-date Tue Apr 25 08:41:25 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_PRef.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:50:40 2017-date Tue Apr 25 08:41:25 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PRef,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PRef --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PRef_DATA.root --fileout file:RelVal_Raw_PRef_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:50:40 2017-date Tue Apr 25 08:41:25 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full2015Data --data --scenario=HeavyIons -n 10 --conditions auto:run2_hlt_HIon --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016,Run2_HI --magField 38T_PostLS1 --fileout file:RelVal_Raw_HIon_DATA.root --filein /store/hidata/HIRun2015/HIHardProbes/RAW-RECO/HighPtJet-PromptReco-v1/000/263/689/00000/1802CD9A-DDB8-E511-9CF9-02163E0138CA.root : FAILED - time: date Tue Apr 25 08:50:20 2017-date Tue Apr 25 08:41:29 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_HIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:50:20 2017-date Tue Apr 25 08:41:29 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:HIon,RAW2DIGI,L1Reco,RECO --data --scenario=HeavyIons -n 10 --conditions auto:run2_data_HIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016,Run2_HI --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_HIon_DATA.root --fileout file:RelVal_Raw_HIon_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:50:20 2017-date Tue Apr 25 08:41:29 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_PIon --relval 9000,50 --datatier "RAW" --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --eventcontent RAW --fileout file:RelVal_Raw_PIon_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 08:50:48 2017-date Tue Apr 25 08:41:48 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_PIon.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:50:48 2017-date Tue Apr 25 08:41:48 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:PIon,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_PIon --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_PIon_DATA.root --fileout file:RelVal_Raw_PIon_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:50:48 2017-date Tue Apr 25 08:41:48 2017 s - exit: 21504
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/Utilities/ReleaseScripts/scripts/read312RV_cfg.py : FAILED - time: date Tue Apr 25 08:49:13 2017-date Tue Apr 25 08:42:00 2017 s - exit: 23552
cmsDriver.py RelVal -s L1REPACK:GT1   --data --scenario=pp -n 10 --conditions auto:run1_hlt_Fake --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --fileout file:RelVal_Raw_Fake_DATA.root --filein /store/data/Run2012A/MuEG/RAW/v1/000/191/718/14932935-E289-E111-830C-5404A6388697.root : FAILED - time: date Tue Apr 25 08:50:25 2017-date Tue Apr 25 08:42:01 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_Fake.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:50:25 2017-date Tue Apr 25 08:42:01 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:Fake,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run1_data_Fake --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --processName=HLTRECO --filein file:RelVal_Raw_Fake_DATA.root --fileout file:RelVal_Raw_Fake_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:50:25 2017-date Tue Apr 25 08:42:01 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_GRun --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2017 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 08:52:08 2017-date Tue Apr 25 08:42:05 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_GRun.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:52:08 2017-date Tue Apr 25 08:42:05 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2017 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun_DATA.root --fileout file:RelVal_Raw_GRun_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:52:08 2017-date Tue Apr 25 08:42:05 2017 s - exit: 21504
cmsDriver.py RelVal -s L1REPACK:Full --data --scenario=pp -n 10 --conditions auto:run2_hlt_GRun2016 --relval 9000,50 --datatier "RAW" --eventcontent RAW --customise=HLTrigger/Configuration/CustomConfigs.L1T --era Run2_2016 --magField 38T_PostLS1 --fileout file:RelVal_Raw_GRun2016_DATA.root --filein /store/data/Run2016B/JetHT/RAW/v1/000/272/762/00000/C666CDE2-E013-E611-B15A-02163E011DBE.root : FAILED - time: date Tue Apr 25 08:52:09 2017-date Tue Apr 25 08:42:13 2017 s - exit: 23552
cmsRun /cvmfs/cms-ib.cern.ch/week1/slc6_amd64_gcc530/cms/cmssw/CMSSW_9_0_X_2017-04-23-0000/src/HLTrigger/Configuration/test/OnLine_HLT_GRun2016.py realData=True  globalTag=@ inputFiles=@  : FAILED - time: date Tue Apr 25 08:52:09 2017-date Tue Apr 25 08:42:13 2017 s - exit: 21504
cmsDriver.py RelVal -s HLT:GRun2016,RAW2DIGI,L1Reco,RECO --data --scenario=pp -n 10 --conditions auto:run2_data_GRun2016 --relval 9000,50 --datatier "RAW-HLT-RECO" --eventcontent FEVTDEBUGHLT --customise=HLTrigger/Configuration/CustomConfigs.L1THLT --era Run2_2016 --magField 38T_PostLS1 --processName=HLTRECO --filein file:RelVal_Raw_GRun2016_DATA.root --fileout file:RelVal_Raw_GRun2016_DATA_HLT_RECO.root : FAILED - time: date Tue Apr 25 08:52:09 2017-date Tue Apr 25 08:42:13 2017 s - exit: 21504

cmsbuild commented:
Comparison not run due to runTheMatrix errors (RelVals and Igprof tests were also skipped)

arunhep commented:
+1
the tests failed because of xrood/eos errors. but they have passed in earlier runs.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next CMSSW_9_0_X IBs (but tests are reportedly failing). This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

ghellwig commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: f1440f81633fa24e88b404e1cd8e2af62244f990
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19468/summary.html

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-18178/19468/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 23
 - DQMHistoTests: Total histograms compared: 1884674
 - DQMHistoTests: Total failures: 29505
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 1854996
 - DQMHistoTests: Total skipped: 173
 - DQMHistoTests: Total Missing objects: 0
 - Checked 94 log files, 14 edm output root files, 23 DQM output files
